### PR TITLE
[KV integrity checker] Remove extra stateUpdates.compare()

### PR DESCRIPTION
**Pull request by fabiotudone-da**
An extra `stateUpdates.compare()` causes state updates to be checked even when in index-only mode.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Comment by fabiotudone-da** [comment]
We should probably add unit tests about config options working properly (like index-only in this case) but, this being an internal tool, not sure about the ROI. Opinions?

**Review by miklos-da — APPROVED** [review]
Yes, adding at least happy path tests for the `indexOnly` option would be great.

File: ledger/participant-state/kvutils/tools/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker [hunk 1]
```diff
@@ -0,0 +1 @@
+mock-maker-inline
```
Comment (fabiotudone-da) [inline ledger/participant-state/kvutils/tools/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker:1]: Enables mocking final classes.

Comment (SamirTalwar-DA) [reply ledger/participant-state/kvutils/tools/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker:1]: Urgh. Can't see a better way around this though.

Perhaps a future implementation might add a trait and make the state updates injectable.

**Review by SamirTalwar-DA — APPROVED** [review]

**Pull request merged**